src/Type.h

```diff
@@ -214,8 +214,8 @@ class Type : public Obj {
     const TableType* AsTableType() const;
     TableType* AsTableType();
 
-    const SetType* AsSetType() const;
-    SetType* AsSetType();
+    const TableType* AsSetType() const;
+    TableType* AsSetType();
```

Comment:
We do have a prior art for not changing the function definitions here, especially since they're not in the detail namespace. I know this goes back a while, but Jon did the same with `Val::Type()` in https://github.com/zeek/zeek/commit/1eb723fc9d6ba8c0bf98353ba958b96d288987f3. I swear @awelzel  did the same in another PR but I can't find it quickly. He might be able to recall more easily. The thing I'm struggling with is that changing the name of the function breaks the naming used by all of the other functions.

Is it possible we could deprecate `AsSetType()` as a whole and just remove it later, instead of changing the return type?